ngraph/frontend/frontend_manager/include/frontend_manager/frontend.hpp

```diff
@@ -26,43 +27,34 @@ namespace ngraph
-            /// frontend implementation is responsible to handle this case, generally frontend may
-            /// retrieve other file names from main file
-            /// \param path Main model file path
-            /// \return Loaded input model
-            virtual InputModel::Ptr load_from_file(const std::string& path) const;
-
-            /// \brief Loads an input model by specified number of model files
-            /// This shall be used for cases when client knows all model files (model, weights, etc)
-            /// \param paths Array of model files
-            /// \return Loaded input model
-            virtual InputModel::Ptr load_from_files(const std::vector<std::string>& paths) const;
-
-            /// \brief Loads an input model by already loaded memory buffer
-            /// Memory structure is frontend-defined and is not specified in generic API
-            /// \param model Model memory buffer
-            /// \return Loaded input model
-            virtual InputModel::Ptr load_from_memory(const void* model) const;
-
-            /// \brief Loads an input model from set of memory buffers
-            /// Memory structure is frontend-defined and is not specified in generic API
-            /// \param modelParts Array of model memory buffers
-            /// \return Loaded input model
-            virtual InputModel::Ptr
-                load_from_memory_fragments(const std::vector<const void*>& modelParts) const;
-
-            /// \brief Loads an input model by input stream representing main model file
-            /// \param stream Input stream of main model
-            /// \return Loaded input model
-            virtual InputModel::Ptr load_from_stream(std::istream& stream) const;
-
-            /// \brief Loads an input model by input streams representing all model files
-            /// \param streams Array of input streams for model
-            /// \return Loaded input model
-            virtual InputModel::Ptr
-                load_from_streams(const std::vector<std::istream*>& streams) const;
+            /// \brief Validates if FrontEnd can recognize model with parameters specified.
+            /// Same parameters should be used to load model.
+            /// \param vars Any number of parameters of any type. What kind of parameters
+            /// are accepted is determined by each FrontEnd individually, typically it is
+            /// std::string containing path to the model file. For more information please
+            /// refer to specific FrontEnd documentation.
+            /// \return true if model recognized, false - otherwise.
+            template <typename... Types>
+            bool supported(const Types&... vars) const
+            {
+                return supported_by_variants({make_variant(vars)...});
+            }
+
+            virtual bool
+                supported_by_variants(const std::vector<std::shared_ptr<Variant>>& variants) const;
```

Comment:
shouldn't we move it to `protected` since it seems no need to have it in public API.

Reply:
supported_by_variants is used by FrontenManager -> Impl, don't think there is a way to do this other than public method.

Reply:
Looks like it is a service method, and I agree with @rkazants it shouldn't be a part of public API.

Or if it is a public method, why don't we have a documentation for it? I am a little bit confused, how should I implement this method in my own framework

Reply:
@slyalin Do we have a plan to refactor API? Now it looks strange that we add service methods to public API